### PR TITLE
upgrade: web3.py version 4.5.0 to 5.7.0

### LINKED ISSUE #6357: Upgrade Web3.py
### User Story

https://web3py.readthedocs.io/en/stable/v5_migration.html#migrating-v4-to-v5

[comment]: # (As a <user type>, I want to <task> so that <goal>.)

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*:

### Current Behavior
[comment]: # (Describe what actually happened.)

### Expected Behavior
[comment]: # (Describe what you expected to happen.)

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)

### Data Requirements
[comment]: # (How will we measure the success of this feature? What kind of tracking is needed for this feature (clicks, impressions, flag)?)

### Additional Information
[comment]: # (Any other information that would be useful, content, screenshots, etc.)

### PR DESCRIPTION
**Pull request by developerfred**
*[ref: migrate guide](https://web3py.readthedocs.io/en/stable/v5_migration.html#migrating-v4-to-v5)

##### Description
upgrade web3 on files:
```
app\dashboard\management\commands\sync_listener.py
app\dashboard\utils.py
app\enssubdomain\views.py
app\event_ethdenver2019\views.py
app\grants\views.py
app\feeswapper\management\commands\feeSwapper.py
app\kudos\management\commands\sync_kudos.py
app\kudos\management\commands\sync_kudos_listener.py
app\kudos\utils.py
app\kudos\views.py
app\townsquare\management\commands\payout_mini_clr.py
```

Create functions
```
- getBlock__, 
- getTransaction__, 
- getTransactionCount__ and 
- getTransactionReceipt___
```

##### Refers/Fixes
 #6357

### CODE REVIEW

**Comment by inspireme6** [comment]
This is a good start, I think there are other changes to be made specifically regarding getBlock, getTransaction, getTransactionCount and getTransactionReceipt. Latest Web3 version returns an exception instead of None.

The files needed to be updated:
```
app\dashboard\management\commands\sync_listener.py
app\dashboard\utils.py
app\enssubdomain\views.py
app\event_ethdenver2019\views.py
app\grants\views.py
app\feeswapper\management\commands\feeSwapper.py
app\kudos\management\commands\sync_kudos.py
app\kudos\management\commands\sync_kudos_listener.py
app\kudos\utils.py
app\kudos\views.py
app\townsquare\management\commands\payout_mini_clr.py
```

An example of code which is going to fail using latest web3 version (dashboard/utils.py):
```python
def has_tx_mined(txid, network):
    web3 = get_web3(network)
    try:
        transaction = web3.eth.getTransaction(txid)
        if not transaction:
            return False
        if not transaction.blockHash:
            return False
        return transaction.blockHash != HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000')
    except Exception:
        return False
```

A solution which is going to avoid adding try/except everywhere might be creating functions using the web3 functions and returning None if they raise an exception. Like this:

```python
def getTransaction__(txid):
    try:
        return web3.eth.getTransaction(txid)
    except TransactionNotFound:
        return None
```

**Comment by developerfred** [comment]
> This is a good start, I think there are other changes to be made specifically regarding getBlock, getTransaction, getTransactionCount and getTransactionReceipt. Latest Web3 version returns an exception instead of None.
> 
> The files needed to be updated:
> 
> ```
> app\dashboard\management\commands\sync_listener.py
> app\dashboard\utils.py
> app\enssubdomain\views.py
> app\event_ethdenver2019\views.py
> app\grants\views.py
> app\feeswapper\management\commands\feeSwapper.py
> app\kudos\management\commands\sync_kudos.py
> app\kudos\management\commands\sync_kudos_listener.py
> app\kudos\utils.py
> app\kudos\views.py
> app\townsquare\management\commands\payout_mini_clr.py
> ```
> 
> An example of code which is going to fail using latest web3 version (dashboard/utils.py):
> 
> ```python
> def has_tx_mined(txid, network):
>     web3 = get_web3(network)
>     try:
>         transaction = web3.eth.getTransaction(txid)
>         if not transaction:
>             return False
>         if not transaction.blockHash:
>             return False
>         return transaction.blockHash != HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000')
>     except Exception:
>         return False
> ```
> 
> A solution which is going to avoid adding try/except everywhere might be creating functions using the web3 functions and returning None if they raise an exception. Like this:
> 
> ```python
> def getTransaction__(txid):
>     try:
>         return web3.eth.getTransaction(txid)
>     except TransactionNotFound:
>         return None
> ```

3d0029e6d..3e07affec

File: app/app/context.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def preprocess(request):
         'header_msg': header_msg,
         'nav_salt': nav_salt,
         'footer_msg': footer_msg,
-        'INFURA_V3_PROJECT_ID': settings.INFURA_V3_PROJECT_ID,
+        'WEB3_INFURA_PROJECT_ID': settings.WEB3_INFURA_PROJECT_ID,
```
Comment (danlipert) [inline app/app/context.py:144]: What does this change do?

File: app/dashboard/tests/test_dashboard_utils.py [hunk 2]
```diff
@@ -297,3 +297,16 @@ def test_release_bounty_to_public_is_successful():
         assert bounty.bounty_reserved_for_user is None
         assert bounty.reserved_for_user_from is None
         assert bounty.reserved_for_user_expiration is None
+
+    def test_getBlock_is_sucess():
```
Comment (danlipert) [inline app/dashboard/tests/test_dashboard_utils.py:301]: some typos here in the method names

File: app/dashboard/utils.py [hunk 3]
```diff
@@ -998,3 +994,38 @@ def list_urls(lis, acc=None):
...
+
+def getTransaction__(txid):
+    """Returns a transaction"""
+    try:
+        return web3.eth.getTransaction(txid)
+    except TransactionNotFound:
+        return None
+
+
+def getBlock__(latest):
```
Comment (danlipert) [inline app/dashboard/utils.py:1010]: I don't understand the naming convention here - what does the trailing double underscore signify?

File: app/dashboard/tests/test_dashboard_utils.py [hunk 4]
```diff
@@ -297,3 +297,16 @@ def test_release_bounty_to_public_is_successful():
...
+        assert getBlock__('latest') is not None
+
+    def test_getBlock_is_BlockNotFound():
```
Comment (danlipert) [inline app/dashboard/tests/test_dashboard_utils.py:304]: What are these tests actually testing? Seems they only test the connection to the web3 provider, which is largely out of our control or merely tests whether the test environment has the proper API keys set up

File: app/dashboard/utils.py [hunk 5]
```diff
@@ -998,3 +994,38 @@ def list_urls(lis, acc=None):
...
+    try:
+        return web3.eth.getBlock(latest)
+    except BlockNotFound:
+        return None
+
+def getTransactionCount__(address):
```
Comment (danlipert) [inline app/dashboard/utils.py:1016]: Why do we want to wrap these web3 calls in try/except blocks? Wouldn't it be better to fail loudly? https://realpython.com/the-most-diabolical-python-antipattern/

File: package-lock.json [hunk 6]
```diff
@@ -199,6 +199,34 @@
         }
       }
     },
+    "@cypress/request": {
```
Comment (danlipert) [inline package-lock.json:202]: lets remove this file from this PR as its unrelated

File: requirements/base.txt [hunk 7]
```diff
@@ -1,7 +1,7 @@
 bs4
```
Comment (danlipert) [inline requirements/base.txt:1]: are these updated pins all necessary for the web3 lib upgrade?

File: app/dashboard/utils.py (hunk 3 above)
Comment (inspireme6) [reply app/dashboard/utils.py:1010]: Regarding name conversion, there's nothing special with it. I added the trailing underscore to differentiate it from the original function when I was testing it, I then copy pasted it here.

File: app/app/context.py (hunk 1 above)
Comment (inspireme6) [reply app/app/context.py:144]: Latest version uses a new enviroment variable.

File: app/dashboard/tests/test_dashboard_utils.py (hunk 2 above)
Comment (developerfred) [reply app/dashboard/tests/test_dashboard_utils.py:301]: I will fix these errors.

File: app/dashboard/utils.py (hunk 5 above)
Comment (developerfred) [reply app/dashboard/utils.py:1016]: @danlipert do you prefer to have an error treatment in each method call?

File: package-lock.json (hunk 6 above)
Comment (developerfred) [reply package-lock.json:202]: ok

File: requirements/base.txt (hunk 7 above)
Comment (developerfred) [reply requirements/base.txt:1]: yes

File: app/dashboard/utils.py (hunk 3 above)
Comment (developerfred) [reply app/dashboard/utils.py:1010]: I will remove the underscore

File: app/dashboard/tests/test_dashboard_utils.py (hunk 4 above)
Comment (developerfred) [reply app/dashboard/tests/test_dashboard_utils.py:304]: @danlipert do you think we don't need these tests?  the web3.py lib has good coverage, if necessary I remove these tests.

Comment (danlipert) [reply app/dashboard/tests/test_dashboard_utils.py:304]: Yeah I think we can remove these

File: app/event_ethdenver2019/views.py [hunk 8]
```diff
@@ -5,7 +5,7 @@
 from django.shortcuts import redirect
```
Comment (danlipert) [inline app/event_ethdenver2019/views.py:5]: We removed this file recently so no need for the changes here

File: package.json [hunk 9]
```diff
@@ -21,7 +21,7 @@
   },
   "homepage": "https://github.com/gitcoinco/web#readme",
   "devDependencies": {
-    "cypress": "^3.4.1",
+    "cypress": "^4.3.0",
```
Comment (danlipert) [inline package.json:24]: no need to update these pins

File: requirements/base.txt [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
 ccxt
 cryptocompare==0.6
-cryptography==2.3
+cryptography==2.6.1
```
Comment (danlipert) [inline requirements/base.txt:4]: these pins as well, if they aren't needed for this PR please revert the change

File: package-lock.json [hunk 11]
```diff
@@ -1,5040 +0,0 @@
-{
```
Comment (danlipert) [inline package-lock.json:1]: no need to delete this

**Comment by thelostone-mc** [comment]
Closing this out now ! 
Will reopen when we reprioritize this (which hopefully is soon)
